### PR TITLE
refactor(ngdocs): Refactor ng docs with updated changes

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA: https://jira.rax.io/browse/FRMW-1263
### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM

### CODE REVIEW

File: src/elements/Tags/scripts/rxTags.js [hunk 1]
```diff
@@ -41,8 +41,8 @@ angular.module('encore.ui.elements')
  *
  * This component can be disabled via the `disabled` attribute or `ng-disabled`
  * directive.
- * @param {Array} options - The list of available tags.
- * @param {String=} [key=undefined] - Determines a value of the tag object to
+ * @param {Array} options The list of available tags.
+ * @param {String=} [key=undefined] Determines a value of the tag object to
```
Comment (CITguy) [inline src/elements/Tags/scripts/rxTags.js]: No need to default value here, optional param syntax should be enough.

```
@param {String=} key ...
```

File: src/elements/Breadcrumbs/scripts/rxBreadcrumbs.js [hunk 2]
```diff
@@ -11,14 +11,14 @@ angular.module('encore.ui.elements')
  * By default, the first breadcrumb will always have an URL of `'/'` and a name of `'Home'`. This can be changed
  * with the `rxBreadcrumbsSvc.setHome` method (see {@link utilities.service:rxBreadcrumbsSvc rxBreadcrumbsSvc}).
  *
- * @param {String=} status
+ * @param {String} status
```
Comment (cathysiller) [inline src/elements/Breadcrumbs/scripts/rxBreadcrumbs.js]: reverted

File: src/elements/Breadcrumbs/scripts/rxBreadcrumbs.js [hunk 3]
```diff
@@ -11,14 +11,14 @@ angular.module('encore.ui.elements')
...
  * The tag to apply to any breadcrumbs with usePageStatusTag:true
  *
  * This leverages the tags defined in {@link rxApp} to display status tags directly inside of breadcrumbs.
  * For a given breadcrumb, `status` will take precedence over `usePageStatusTag`, i.e. it will use a tag defined in
  * `status` instead of checking for and using a tag for the page.
  *
- * @param {Boolean=} [usePageStatusTag=false]
+ * @param {Boolean=} usePageStatusTag=false
```
Comment (cathysiller) [inline src/elements/Breadcrumbs/scripts/rxBreadcrumbs.js]: reverted

File: src/components/rxApp/scripts/rxAppNavItem.js [hunk 4]
```diff
@@ -65,7 +65,7 @@ angular.module('encore.ui.rxApp')
              * _This is *NOT* meant for general consumption, this is strictly for the Origin Project_
              * _This will eventually be deprecated and removed_
              *
-             * @param {string} [url] - URL for the nav item's href
+             * @param {string=} url - URL for the nav item's href
```
Comment (cathysiller) [inline src/components/rxApp/scripts/rxAppNavItem.js]: check
check

File: src/components/rxApp/scripts/rxAppNavItem.js [hunk 5]
```diff
@@ -132,9 +132,9 @@ angular.module('encore.ui.rxApp')
             /*
              * @description Determines whether or not a nav item should be displayed, based on `visibility`
              * criteria and `roles` criteria
-             * @param [visibility] - Can be an expression, a function, an array (using format below) to
+             * @param {object} visibility Can be an expression, a function, an array (using format below) to
```
Comment (cathysiller) [inline src/components/rxApp/scripts/rxAppNavItem.js]: check

File: src/components/rxApp/scripts/rxAppNavItem.js [hunk 6]
```diff
@@ -132,9 +132,9 @@ angular.module('encore.ui.rxApp')
...
              *                     determine visibility
-             * @param {object} [roles] - An object with a format { 'any': ['role1', 'role2'] } or
+             * @param {object=} roles An object with a format { 'any': ['role1', 'role2'] } or
```
Comment (cathysiller) [inline src/components/rxApp/scripts/rxAppNavItem.js]: check

File: src/components/rxOptionTable/scripts/rxOptionTable.js [hunk 7]
```diff
@@ -8,11 +8,11 @@ angular.module('encore.ui.rxOptionTable')
  * @description
  * This directive is used to build a table of radio/checkbox inputs.
  *
- * @param {String} field-id - Used as the base for unique identifiers within the generated markup.
- * @param {Object} model - The AngularJS model to tie all radios/checkboxes together.
- * @param {String} type - (`radio` | `checkbox`) Type of input to be used.
- * @param {String} empty-message - A default message if the data attribute is empty.
- * @param {Array} data - Array of objects used to populate table. Properties must match column keys.
+ * @param {String} field-id Used as the base for unique identifiers within the generated markup.
```
Comment (cathysiller) [inline src/components/rxOptionTable/scripts/rxOptionTable.js]: check

File: src/components/rxOptionTable/scripts/rxOptionTable.js [hunk 8]
```diff
@@ -8,11 +8,11 @@ angular.module('encore.ui.rxOptionTable')
...
+ * @param {Object} model The AngularJS model to tie all radios/checkboxes together.
+ * @param {String} type (`radio` | `checkbox`) Type of input to be used.
+ * @param {String} empty-message A default message if the data attribute is empty.
```
Comment (cathysiller) [inline src/components/rxOptionTable/scripts/rxOptionTable.js]: check

File: src/components/rxOptionTable/scripts/rxOptionTable.js [hunk 9]
```diff
@@ -45,11 +45,11 @@ angular.module('encore.ui.rxOptionTable')
...
  * }]
  * </pre>
- * @param {String=} selected - Array of objects to match against data for preselection on page load.
+ * @param {String=} selected Array of objects to match against data for preselection on page load.
  * If excluded, no values will be preselected on initial load.
- * @param {Boolean=} required - Value passed to input's `ng-required` attribute.
+ * @param {Boolean=} required Value passed to input's `ng-required` attribute.
  * For checkboxes, a `true` value means that there must be at least one checkbox selected.
- * @param {Function=} disable-fn - Optional callback function to determine if option should be disabled.
+ * @param {Function=} disable-fn Optional callback function to determine if option should be disabled.
```
Comment (cathysiller) [inline src/components/rxOptionTable/scripts/rxOptionTable.js]: check

File: src/elements/Collapse/scripts/rxCollapse.js [hunk 10]
```diff
@@ -21,7 +21,7 @@ angular.module('encore.ui.elements')
  * get lost in between metadata key/value pairs.
  *
  *
- * @param {String=} [title="See More/See Less"]
+ * @param {String=} title="See More/See Less"
```
Comment (cathysiller) [inline src/elements/Collapse/scripts/rxCollapse.js]: reverted

File: src/elements/Forms/scripts/rxCharacterCount.js [hunk 11]
```diff
@@ -63,8 +63,8 @@ angular.module('encore.ui.elements')
  * * Content that causes a scrollbar in the textarea
  * * Initial text (coming from the model) that is over the limit
  *
- * @param {Number=} [low-boundary=10] How far from the maximum to enter a warning state
- * @param {Number=} [max-characters=254] The maximum number of characters allowed
+ * @param {Number} low-boundary=10 How far from the maximum to enter a warning state
```
Comment (cathysiller) [inline src/elements/Forms/scripts/rxCharacterCount.js]: reverted

File: src/elements/Forms/scripts/rxCharacterCount.js [hunk 12]
```diff
@@ -63,8 +63,8 @@ angular.module('encore.ui.elements')
...
+ * @param {Number} max-characters=254 The maximum number of characters allowed
```
Comment (cathysiller) [inline src/elements/Forms/scripts/rxCharacterCount.js]: reverted

File: src/elements/Forms/scripts/rxFieldName.js [hunk 13]
```diff
@@ -41,7 +41,7 @@ angular.module('encore.ui.elements')
  * ...
  * </pre>
  *
- * @param {Boolean=} [ng-required=false]
+ * @param {Boolean} ng-required=false
```
Comment (cathysiller) [inline src/elements/Forms/scripts/rxFieldName.js]: updated

File: src/elements/Forms/scripts/rxToggleSwitch.js [hunk 14]
```diff
@@ -26,8 +26,8 @@ angular.module('encore.ui.elements')
  * @param {Function} postHook A function to run when the switch is toggled
  * @param {Expression=} [ngDisabled=false] If the expression is truthy, then the
  * `disabled` attribute will be set on the toggle switch.
- * @param {Expression=} [trueValue=true] The value of the scope property when the switch is on
- * @param {Expression=} [falseValue=false] The value of the scope property when the switch is off
+ * @param {Expression=} trueValue=true The value of the scope property when the switch is on
```
Comment (cathysiller) [inline src/elements/Forms/scripts/rxToggleSwitch.js]: reverted

File: src/elements/Forms/scripts/rxToggleSwitch.js [hunk 15]
```diff
@@ -26,8 +26,8 @@ angular.module('encore.ui.elements')
...
+ * @param {Expression=} falseValue=false The value of the scope property when the switch is off
```
Comment (cathysiller) [inline src/elements/Forms/scripts/rxToggleSwitch.js]: reverted

File: src/elements/Modals/scripts/rxModalForm.js [hunk 16]
```diff
@@ -16,10 +16,10 @@ angular.module('encore.ui.elements')
  * @param {String} title Title of modal window
  * @param {String=} subtitle Subtitle of modal window
  * @param {Boolean=} isLoading True to show a spinner by default
- * @param {String=} [submitText='Submit'] 'Submit' button text to use.
- * @param {String=} [cancelText='Cancel'] 'Cancel' button text to use.
- * @param {String=} [returnText='Return'] 'Return' button text to use.
- * @param {String=} defaultFocus
+ * @param {String=} submitText='Submit' 'Submit' button text to use.
```
Comment (cathysiller) [inline src/elements/Modals/scripts/rxModalForm.js]: reverted

File: src/elements/Modals/scripts/rxModalForm.js [hunk 17]
```diff
@@ -16,10 +16,10 @@ angular.module('encore.ui.elements')
...
+ * @param {String=} cancelText='Cancel' 'Cancel' button text to use.
```
Comment (cathysiller) [inline src/elements/Modals/scripts/rxModalForm.js]: reverted

File: src/elements/Modals/scripts/rxModalForm.js [hunk 18]
```diff
@@ -16,10 +16,10 @@ angular.module('encore.ui.elements')
...
+ * @param {String=} returnText='Return' 'Return' button text to use.
```
Comment (cathysiller) [inline src/elements/Modals/scripts/rxModalForm.js]: reverted

File: src/elements/Modals/scripts/rxModalForm.js [hunk 19]
```diff
@@ -16,10 +16,10 @@ angular.module('encore.ui.elements')
...
+ * @param {String=} [defaultFocus]
```
Comment (cathysiller) [inline src/elements/Modals/scripts/rxModalForm.js]: reverted

File: src/elements/Notifications/scripts/rxNotification.js [hunk 20]
```diff
@@ -7,7 +7,7 @@ angular.module('encore.ui.elements')
  * @description
  * Display a static message with styling taken from `rx-notifications`.
  *
- * @param {String=} [type='info'] The type of notification (e.g. 'warning', 'error')
+ * @param {String} [type='info'] The type of notification (e.g. 'warning', 'error')
```
Comment (cathysiller) [inline src/elements/Notifications/scripts/rxNotification.js]: reverted

File: src/utilities/rxAutoSave/scripts/rxAutoSave.js [hunk 21]
```diff
@@ -286,7 +286,7 @@ angular.module('encore.ui.utilities')
  * @param {String} variable
  * variable name corresponding to an object on the given scope
  * @param {Object=} options usage options
- * @param {Promise=} [options.clearOnSuccess=null]
+ * @param {Promise=} options.clearOnSuccess=null
```
Comment (cathysiller) [inline src/utilities/rxAutoSave/scripts/rxAutoSave.js]: updated

File: src/utilities/rxAutoSave/scripts/rxAutoSave.js [hunk 22]
```diff
@@ -324,11 +324,11 @@ angular.module('encore.ui.utilities')
  * * `clear()`
  * * `clearOnSuccess()`
  *
- * @param {Boolean|Promise=} [options.load=true]
+ * @param {Boolean|Promise=} options.load=true
```
Comment (cathysiller) [inline src/utilities/rxAutoSave/scripts/rxAutoSave.js]: reverted

File: src/utilities/rxAutoSave/scripts/rxAutoSave.js [hunk 23]
```diff
@@ -324,11 +324,11 @@ angular.module('encore.ui.utilities')
...
  * If false, will prevent data from being automatically loaded onto the scope.
  *
  * You may use a promise that resolves to a boolean, if desired.
- * @param {Boolean|Promise=} [options.save=true]
+ * @param {Boolean|Promise=} options.save=true
```
Comment (cathysiller) [inline src/utilities/rxAutoSave/scripts/rxAutoSave.js]: reverted

File: src/utilities/rxAutoSave/scripts/rxAutoSave.js [hunk 24]
```diff
@@ -342,7 +342,7 @@ angular.module('encore.ui.utilities')
  * });
  * </pre>
  *
- * @param {Object=} [options.storageBackend=rxLocalStorage]
+ * @param {Object=} options.storageBackend=rxLocalStorage
```
Comment (cathysiller) [inline src/utilities/rxAutoSave/scripts/rxAutoSave.js]: reverted

File: src/utilities/rxEnvironment/scripts/rxEnvironment.js [hunk 25]
```diff
@@ -189,7 +189,7 @@ angular.module('encore.ui.utilities')
     /*
      * Retrieves current environment
      * @public
-     * @param {string} [href] The path to check the environment on. Defaults to $location.absUrl()
+     * @param {string} href The path to check the environment on. Defaults to $location.absUrl()
```
Comment (cathysiller) [inline src/utilities/rxEnvironment/scripts/rxEnvironment.js]: updated

File: src/utilities/rxEnvironment/scripts/rxEnvironment.js [hunk 26]
```diff
@@ -245,8 +245,8 @@ angular.module('encore.ui.utilities')
      * Given an environment name, check if any of our registered environments
      * match it
      * @public
-     * @param {string} [name] Environment name to check
-     * @param {string} [href] Optional href to check against. Defaults to $location.absUrl()
+     * @param {string} name Environment name to check
```
Comment (cathysiller) [inline src/utilities/rxEnvironment/scripts/rxEnvironment.js]: updated

File: src/utilities/rxEnvironment/scripts/rxEnvironment.js [hunk 27]
```diff
@@ -245,8 +245,8 @@ angular.module('encore.ui.utilities')
...
+     * @param {string=} [href] Optional href to check against. Defaults to $location.absUrl()
```
Comment (cathysiller) [inline src/utilities/rxEnvironment/scripts/rxEnvironment.js]: updated

File: src/utilities/rxNestedElement/scripts/rxNestedElement.js [hunk 28]
```diff
@@ -7,7 +7,7 @@ angular.module('encore.ui.utilities')
  * required to validate nested custom elements.
  *
  * @param {Object=} opts - Options to merge with default DDO definitions
```
Comment (cathysiller) [inline src/utilities/rxNestedElement/scripts/rxNestedElement.js]: updated

File: src/utilities/rxNotify/scripts/rxNotify.js [hunk 29]
```diff
@@ -331,9 +331,9 @@ angular.module('encore.ui.utilities')
      * Values:
      * * -1
      * * Any positive integer
-     * @param {Boolean=} [options.repeat=true]
+     * @param {Boolean=} options.repeat=true
```
Comment (cathysiller) [inline src/utilities/rxNotify/scripts/rxNotify.js]: updated

File: src/utilities/rxNotify/scripts/rxNotify.js [hunk 30]
```diff
@@ -331,9 +331,9 @@ angular.module('encore.ui.utilities')
...
      * Whether the message should be allowed to appear more than once in the stack.
-     * @param {Boolean=} [options.loading=false]
+     * @param {Boolean=} options.loading=false
```
Comment (cathysiller) [inline src/utilities/rxNotify/scripts/rxNotify.js]: updated

File: src/utilities/rxNotify/scripts/rxNotify.js [hunk 31]
```diff
@@ -346,17 +346,17 @@ angular.module('encore.ui.utilities')
      * * [scope, 'property']
      *   * Pass in a property on a scope to watch for a change.
      *     When the property equals true, the message is shown.
-     * @param {String|Array=} [options.dismiss='next']
+     * @param {String|Array=} options.dismiss='next'
```
Comment (cathysiller) [inline src/utilities/rxNotify/scripts/rxNotify.js]: reverted

File: src/utilities/rxNotify/scripts/rxNotify.js [hunk 32]
```diff
@@ -346,17 +346,17 @@ angular.module('encore.ui.utilities')
...
      * When to have the message disappear.
      *
      * Values:
      * * 'next'
      * * [scope, 'property']
      *     * Pass in a property on a scope to watch for a change.
      *       When the property equals true, the message is dismissed.
-     * @param {Function=} [options.ondismiss=_.noop]
+     * @param {Function=} options.ondismiss=_.noop
```
Comment (cathysiller) [inline src/utilities/rxNotify/scripts/rxNotify.js]: reverted

File: src/utilities/rxNotify/scripts/rxNotify.js [hunk 33]
```diff
@@ -346,17 +346,17 @@ angular.module('encore.ui.utilities')
...
      * Function that should be run when message is dismissed.
-     * @param {String=} [options.stack='page']
+     * @param {String=} options.stack='page'
```
Comment (cathysiller) [inline src/utilities/rxNotify/scripts/rxNotify.js]: reverted

File: src/elements/Forms/docs/examples/forms.intermediatecontrols.html [hunk 34]
```diff
@@ -1,4 +1,4 @@
-<form rx-form ng-controller="formIntermediateControlsDemoCtrl" name="formIntermediateControls">
+<form rx-form ng-controller="rxformIntermediateControlsDemoCtrl" name="rxformIntermediateControls">
```
Comment (cathysiller) [inline src/elements/Forms/docs/examples/forms.intermediatecontrols.html]: reverted

File: src/elements/Tags/scripts/rxTags.js (hunk 1 above)
Comment (cathysiller) [inline src/elements/Tags/scripts/rxTags.js]: updated

Comment (cathysiller) [inline src/elements/Tags/scripts/rxTags.js]: grunt gives an error to key=undefined
grunt error goes away with [key=undefined] 
updated to [key=undefined]

Comment (cathysiller) [inline src/elements/Tags/scripts/rxTags.js]: updated with just "key"

**Comment by cynprudence** [comment]
LGTM

**Pull request merged**